src/admin/python/cli/script.py

```diff
@@ -0,0 +1,660 @@
+
+    shutil.copytree(src=dashboard_files_src, dst=dashboard_files_dst)
+    with open(dashboard_launch_gate, "w") as f:
+        pass
+    print(
+        f"Copied dashboard files to {dashboard_files_dst} and wrote the nginx controller config to {nginx_controller_cfg_path}; exiting."
+    )
+
+
+def enter_manager_cmsd(cms_delay_servers):
+    # TODO xrootd_managers needs to be passed in by execution env or mounted in /config-etc
+    save_template_cfg(
+        dict(
+            xrootd_managers=["localhost"],
+            cmsd_manager="UNUSED",
+            cms_delay_servers=cms_delay_servers,
+        )
+    )
+    apply_template_cfg_file(cmsd_manager_cfg_template, cmsd_manager_cfg_path)
+    args = [
+        "cmsd",
+        "-c",
+        cmsd_manager_cfg_path,
+        "-n",
+        "manager",
+        "-I",
+        "v4",
+    ]
+    sys.exit(_run(args))
+
+
+def enter_xrootd_manager(cmsd_manager):
+    save_template_cfg(
+        dict(
+            cmsd_manager=cmsd_manager,
+        )
+    )
+    apply_template_cfg_file(xrootd_manager_cfg_template, xrootd_manager_cfg_path)
+    _run(
+        [
+            "xrootd",
+            "-c",
+            xrootd_manager_cfg_path,
+            "-n",
+            "manager",
+            "-I",
+            "v4",
+        ],
+    )
+
+
+def enter_worker_cmsd(cmsd_manager, vnid, debug_port, connection):
+    parsed = urlparse(connection)
+    save_template_cfg(
+        dict(
+            vnid=vnid,
+            cmsd_manager=cmsd_manager,
+            db_host=parsed.hostname,
+            db_port=parsed.port,
+            mysqld_user_qserv=parsed.username,
+        )
+    )
+
+    apply_template_cfg_file(cmsd_worker_cfg_template, cmsd_worker_cfg_path)
+    apply_template_cfg_file(xrdssi_cfg_template, xrdssi_cfg_path)
+
+    _do_smig_block(admin_smig_dir, "admin", connection)
+
+    args = [
+        "cmsd",
+        "-c",
+        cmsd_worker_cfg_path,
+        "-n",
+        "worker",
+        "-I",
+        "v4",
+        "-l",
+        "@libXrdSsiLog.so",
+        "-+xrdssi",
+        xrdssi_cfg_path,
+    ]
+    sys.exit(_run(args, debug_port=debug_port))
+
+
+def enter_worker_xrootd(
+    debug_port,
+    xrd_port,
+    connection,
+    vnid,
+    cmsd_manager,
+    repl_ctl_dn,
+    mysql_monitor_password,
+    db_qserv_user,
+):
+    """Entrypoint script for the xrootd container that runs on worker nodes."""
+
+    # TODO This sets the amount of data that can be locked into memory to
+    # almost the entire amount of memory on the machine. I think in a dev-env
+    # (docker-only on a single machine) we don't want to grab quite so much
+    # memory? TBD WTD here - do/don't set this? set it to how much? needs a way
+    # to set env (dev/prod/etc)
+    # # Increase limit for locked-in-memory size
+    # MLOCK_AMOUNT=$(grep MemTotal /proc/meminfo | awk '{printf("%.0f\n", $2 - 1000000)}')
+    # ulimit -l "$MLOCK_AMOUNT"
```

Comment:
What is the correct value for this: in production, on a developer machine?
I still need a way to pass it in via cfg I think.